### PR TITLE
Add versions summary to top of README

### LINKED ISSUE #62: Add summary over versions of libraries
Inspired by my inability to react to @rly 's comment during nwb/dandi meeting that a new matnwb was released fixing some issues.  I think that while composing our detailed per dandiset table we should prepend it with summary over versions of libraries. E.g.

```markdown
 ## Versions (good/bad/not-tested):

-  hdmf: 3.11.0 (900/100/10000), 3.10.0 (100/800/10100), ...
-  matnwb: v2.6.0.2 (0/10000/100), ...
-  pynwb: 2.5.0 (11000/0/0)
```

that would give us a nice quick idea on how much testing was done for the recent versions. 

bonus feature: those numbers could be hyperlinks to a sample file which passed/failed (not so interesting for not-tested I guess) for the corresponding version.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #62.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,10 +44,10 @@
...
 | [000035](results/000035/status.yaml) | [184 passed](results/000035/status.yaml#L9), 0 failed, [1 timed out](results/000035/status.yaml#L686) | [2 passed](results/000035/status.yaml#L777), [59 failed](results/000035/status.yaml#L693), [124 timed out](results/000035/status.yaml#L788) | — |
-| [000036](results/000036/status.yaml) | [57 passed](results/000036/status.yaml#L9), 0 failed, 0 timed out | 0 passed, [28 failed](results/000036/status.yaml#L261), [29 timed out](results/000036/status.yaml#L391) | — |
-| [000037](results/000037/status.yaml) | [47 passed](results/000037/status.yaml#L9), 0 failed, 0 timed out | 0 passed, [40 failed](results/000037/status.yaml#L243), [7 timed out](results/000037/status.yaml#L441) | — |
+| [000036](results/000036/status.yaml) | [57 passed](results/000036/status.yaml#L9), 0 failed, 0 timed out | 0 passed, [29 failed](results/000036/status.yaml#L261), [28 timed out](results/000036/status.yaml#L396) | — |
+| [000037](results/000037/status.yaml) | [47 passed](results/000037/status.yaml#L9), 0 failed, 0 timed out | 0 passed, [41 failed](results/000037/status.yaml#L243), [6 timed out](results/000037/status.yaml#L446) | — |
```
Comment (yarikoptic) [inline README.md:48]: results files didn't change but stats changed here, how come? (also in other spots)

Comment (jwodder) [reply README.md:48]: The README wasn't updated when a bunch of `results/` files were last updated in 2c6242c7.

Comment (yarikoptic) [reply README.md:48]: ah cool. Just wanted to make sure that code is not having some "flaky" point

**Comment by yarikoptic** [comment]
Thank you!

**Pull request merged**